client/src/main/java/io/pravega/client/batch/SegmentRange.java

```diff
@@ -60,4 +62,10 @@ public interface SegmentRange extends Serializable {
      * @return This
      */
     SegmentRangeImpl asImpl();
+
+    /**
+     * Returns the segment.
+     * @return The segment.
+     */
+    Segment getASegment();
```

Comment:
```suggestion
    Segment getSegment();
```

Reply:
> 

This can't be modified to getSegment as SegmentRangeImpl already has a field called segment with access level restricted to package like @Getter(value = AccessLevel.PACKAGE). So when adding the method name as getSegment, it calls the getter of the above field and it throws the below error:  **attempting to assign weaker access privileges; was public.**
Either we need to add a new field like we are doing now, or we need to change the access levels.
please suggest.

Reply:
Change the access level